src/util/antoraSupport.ts

```diff
@@ -0,0 +1,34 @@
+import vscode, { WorkspaceFolder } from 'vscode'
+import fs from 'fs'
+import yaml from 'js-yaml'
+
+export default class AntoraSupport {
+  public async isDetected (): Promise<boolean> {
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 10)
+    return antoraYmlFiles.length > 0
+  }
+
+  public findCurrentWorkspaceFolder (): WorkspaceFolder {
+    return vscode.workspace.getWorkspaceFolder(vscode.Uri.file(vscode.window.activeTextEditor.document.uri.fsPath))
+  }
+
+  public findModulesDirPath (): string | undefined {
+    const startPath = vscode.window.activeTextEditor.document.uri.fsPath
+    // check that we stay in the current workspace
+    const currentWorkspaceFolderName = this.findCurrentWorkspaceFolder().name
+    if (startPath.indexOf(currentWorkspaceFolderName) < startPath.indexOf('modules')) {
```

Comment:
That's not reliable, what if we have a workspace named `foo` and the following path: `/path/to/foo/docs/modules/ROOT/pages/foo/page.adoc`?